processing/src/main/java/org/apache/carbondata/processing/store/CarbonFactDataHandlerColumnar.java

```diff
@@ -371,8 +371,13 @@ private void setWritingConfiguration() throws CarbonDataWriterException {
     this.pageSize = Integer.parseInt(CarbonProperties.getInstance()
         .getProperty(CarbonCommonConstants.BLOCKLET_SIZE,
             CarbonCommonConstants.BLOCKLET_SIZE_DEFAULT_VAL));
+    // support less than 32000 rows in one page, because we support super long string,
+    // if it is long enough, a clomun page with 32000 rows will exceed 2GB
     if (version == ColumnarFormatVersion.V3) {
-      this.pageSize = CarbonV3DataFormatConstants.NUMBER_OF_ROWS_PER_BLOCKLET_COLUMN_PAGE_DEFAULT;
+      this.pageSize =
```

Comment:
how much is the default value for page size ?

Reply:
In V3, it is 32000 by default. Here we use the min(32000, user_specified)